standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -6452,22 +6454,42 @@ public GetOpenTxnsInfoResponse get_open_txns_info() throws TException {
 
     @Override
     public OpenTxnsResponse open_txns(OpenTxnRequest rqst) throws TException {
-      return getTxnHandler().openTxns(rqst);
+      OpenTxnsResponse response = getTxnHandler().openTxns(rqst);
+      List<Long> txnIds = response.getTxn_ids();
+      if (txnIds != null && listeners != null && !listeners.isEmpty()) {
+        MetaStoreListenerNotifier.notifyEvent(listeners, EventType.OPEN_TXN,
```

Comment:
How this behaves as we don't have dbConn and sqlGenerator objects here? This is another argument to have 2 different interfaces for direct SQL and DN.